Environment setup

chunchunmaru34 opened the pull request:
Setted up react js environment with express hot-reload server. 
Configs were separated to files according to NODE_ENV value ("production" or "development").
Added React Router and PostCSS.

AliakseiLabkovich committed:
Environment set up including a few main React components

AliakseiLabkovich committed:
a little bit refactoring

AliakseiLabkovich committed:
-Separated containers from presentations components.
-Removed func.bind expressions for more clear code.
-Added config for storing api paths.

MikhailMartsinovich commented on conf/api-paths.js:
```diff
@@ -0,0 +1,2 @@
+export const API = "http://localhost:3003";
+export const MOVIES_PATH = "http://localhost:3003";
```
В чем разница между API и MOVIES_PATH?
И почему они в одном домене и заассайнены на один порт?

MikhailMartsinovich commented on server.js:
```diff
@@ -0,0 +1,35 @@
+const app = express();
+
+
+const PORT = 3000;
```
Храни порт в конфиге

MikhailMartsinovich commented on server.js:
```diff
@@ -0,0 +1,35 @@
+const PORT = 3000;
+const PUBLIC_PATH = __dirname + '/public';
+
+const isDev = process.env.NODE_ENV === 'development';
```
Вынеси определение значения IsDev в отдельный модуль и экспортируй из него это значение.

MikhailMartsinovich commented on src/components/containers/movie-detail/index.js:
```diff
@@ -0,0 +1,26 @@
+            .then(res => this.setState({ movie: res }))
+    };
+    render() {
+        return React.createElement(MovieDetail, { movie: this.state.movie })
```
Зачем здесь CreateElement? лучше использовать JSX

MikhailMartsinovich commented on src/components/presentations/movie/styles.css:
```diff
@@ -0,0 +1,19 @@
+    line-height: 50px;
+    text-decoration: none;
+    background-color: #222222;
+    color: #D4D4D4;
```
Вынеси цвета проекта в отдельный файл с переменными типа variables или colors, Экспортируй в файлы стилей, где нужны цвета и используй эти переменные.

И так для всех цветов по проекту

MikhailMartsinovich reviewed (changes requested):
1) Для каждого компонента, если он принимает хоть какие-то props, обязательно объявляй PropTypes!!!
2) лучше пиши Container в именовании файлов контейнеров. Например movie-detail-container.
3) настрой ESLint и не забывай ставить ';'. 

В Понедельник поговорим об организации проекта, т к иметь 2 папки containers и presentations не очень удобно и нет функционального разделения  компонентов.

AliakseiLabkovich committed:
-Fixed duplicate path in api-paths
-Added server config
-Moved PORT and isDev variables to server config
-Added PropTypes checking
-Removed components separation by their role

AliakseiLabkovich committed:
Code style improvements with ESLint

AliakseiLabkovich committed:
Implementing color constants

MikhailMartsinovich commented on src/components/header/styles.scss:
```diff
@@ -0,0 +1,16 @@
+
+.header {
+    background-color: $dark;
+    color: whitesmoke;
```
Добавь цвет с global.scss

MikhailMartsinovich commented on src/components/movie/movie-card/styles.scss:
```diff
@@ -0,0 +1,20 @@
+.container {
+    display: inline-block;
+    margin: 10px;
+    border: rgba(34,34,34, 0.5) 1px solid;
```
цвет в global.scss

MikhailMartsinovich commented on src/components/movie/movie-detail/movie-detail.js:
```diff
@@ -0,0 +1,53 @@
+
+MovieDetail.propTypes = {
+  movie: PropTypes.shape({
+    _id: PropTypes.string.isRequired,
```
как правило, начиная с "_" именуются приватные члены. Замени лучше просто на id

MikhailMartsinovich commented on src/components/movie/movie-detail/styles.scss:
```diff
@@ -0,0 +1,39 @@
+    text-align: center;
+}
+
+.details {
```
ты не используешь возможности sass. Он позваляет создавать иерархичную структуру. ВОт пример:
.details {
 /*smth*/

 & > div {
    /*smth*/
 }

  .another-class {
    /*smth*/
  }
}

AliakseiLabkovich committed:
Minor style adjustments and refactoring with SASS

AliakseiLabkovich committed:
Changed _id to id
